### ISSUE TITLE
Support AML metrics logging in the shrike.compliant_logging

### ISSUE THREAD

**Issue by haozhestat** [Azure/shrike#84]
Examples in some partner teams:

1. @felixnext 's [logger.py](https://eemo.visualstudio.com/TEE/_git/TEEGit?path=%2FOffline%2FConversationalRelevance%2Fpython%2Flibraries%2Fcr-ds-core%2Fcr_ds_core%2Futils%2Flogger.py&version=GBdevelop)
2. @danrsc 's [Pull Request 22485: Add AML metric logging utilities, convenience method for registering metadata...](https://eemo.visualstudio.com/TEE/_git/TEEGit/pullrequest/22485?_a=files&path=%2FOffline%2Faml_graph_node%2Fsrc%2Faml_graph_node%2Fcommon%2Faml_logger.py)
3. [metric_logging.py](https://eemo.visualstudio.com/TEE/_git/TEEGit?version=GBcompliant%2FPersonalizedViewTime&path=%2FOffline%2FPersonalizedViewTime%2Fpvt%2Futils%2Fmetric_logging.py) in the PersonalViewTime repo.

**Comment by haozhestat** [comment]
After this issue is resolved, we need to refactor the metric logging and compliant logging in the accelerator repo so that they could be unified.

A duplicated workitem in the azure devops:

[Task 1108795: [Iteration #3] Refactor the confidential logger in the accelerator repo](https://dev.azure.com/msdata/Vienna/_queries/edit/1108795/?triage=true)

**Comment by haozhestat** [comment]
@dkmiller, @dansrc and @felixnext : As you know, currently `shrike.compliant_logging` doesn't have any dependencies except the standard python libraries. To enable aml metric logging, we need have "azureml-core" installed in the environment at least. So, I will add "azureml-core" into its requirement list.

Two weeks ago, when  we discussed with @felixnext, we were concerned about the difficulty with installing "azureml-core" in the Heron HDI environment. However, since "azureml-core" was added into HDI by @dkmiller last week, this concern should be mitigated.
